cdap-security/src/main/java/io/cdap/cdap/security/authorization/DefaultAccessEnforcer.java

```diff
@@ -128,6 +135,11 @@ private void doEnforce(EntityId entity, Principal principal, Set<? extends Permi
     if (isAccessingSystemNSAsMasterUser(entity, principal) || isEnforcingOnSamePrincipalId(entity, principal)) {
       return;
     }
+
+    // When user credential encryption is enabled, credential should be encrypted upon arrival
+    // at router and decrypted right here before calling auth extension.
+    principal = decryptUserCredential(principal);
```

Comment:
Similarly, call this `getUserPrincipal(principal)`

Reply:
Done